android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -768,6 +769,11 @@ protected void onSaveInstanceState(Bundle outState)
         fragment.saveAltitudeChartState(outState);
     }
 
+    if(mNavigationController != null)
+    {
+      outState.putBoolean(STATE_SHOW_TIME_LEFT, mNavigationController.saveState());
```

Comment:
@fijemax  please take a look at the 762 line `mRoutingPlanInplaceController.onSavedState(outState)`, i.e. just pass the whole bundle, `outState`, to the `mNavigationController.saveState(@NonNulle Bundle outState)`  to let the navigation controller to save its state as it wants.

Also, please rename the `mNavigationController.saveState()` method to `mNavigationController.onSaveState(@NonNull Bundle outState)` so that to be consistent with other code design/style.